File: source/WeddingManager.Services/PaymentService.cs
using System.Collections.Generic;
using WeddingManager.Core.Data;
using WeddingManager.Core.Repositories;
using WeddingManager.Core.Services;

namespace WeddingManager.Services
{
    public class PaymentService : IPaymentService
    {
        public PaymentService(IPaymentRepository paymentRepository)
        {
            _paymentRepository = paymentRepository;
        }

        public int CreatePayment(int serviceId, Payment payment)
        {
            return _paymentRepository.CreatePayment(serviceId, payment);
        }

        public IEnumerable<Payment> RetrievePayments(int serviceId)
        {
            return _paymentRepository.RetrievePayments(serviceId);
        }

        public void UpdatePayment(Payment payment)
        {
            _paymentRepository.UpdatePayment(payment);
        }

        public void DeletePayment(int paymentId)
        {
            _paymentRepository.DeletePayment(paymentId);
        }

        private readonly IPaymentRepository _paymentRepository;
    }
}

File: source/WeddingManager.Web/js/services/paymentService.js
(function(){
    angular.module('weddingManager')
    .service('paymentService', function(paymentFactory, appStateService){
        var service = {};
        var _payments = [];
        
        service.createPayment = function(serviceId, onSuccess, onError){
            var payment = {
                Method : 1,
                Notes : "New payment",
                DateReceived : new Date()
            };
            paymentFactory.createPayment(serviceId, payment, function(paymentId){
                payment.Id = paymentId;
                onSuccess(payment);
            }, onError);
        }
        
        service.refreshPayments = function(serviceId, onSuccess, onError){
            paymentFactory.retrievePayments(serviceId, function(payments){
                _payments = payments;
                onSuccess(_payments);
            }, onError);
        }
        
        service.updatePayment = function(payment, onSuccess, onError){
            paymentFactory.updatePayment(payment, function(){
                onSuccess();
            }, onError);
        }
        
        service.deletePayment = function(paymentId, onSuccess, onError){
            paymentFactory.deletePayment(paymentId, function(){
                appStateService.setPayment(null);
                onSuccess();
            }, onError);
        }
        
        return service;
    });
})();